Signal/src/environment/OWS106EnsureProfileComplete.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+class OWS106EnsureProfileComplete: OWSDatabaseMigration {
+
+    let TAG = "[OWS106EnsureProfileComplete]"
+
+    // increment a similar constant for each migration.
+    class func migrationId() -> String {
+        return "106"
+    }
+
+    // Overriding runUp since we have some specific completion criteria which
+    // is more likely to fail since it involves network requests.
+    override func runUp() {
+        CompleteRegistrationFixerJob(completionHandler: {
+            Logger.info("\(self.TAG) Completed. Saving.")
+            self.save()
+        }).start()
+    }
```

Comment:
Won't CompleteRegistrationFixerJob get deallocated as soon as it starts?  Who maintains a strong reference to it while it "runs"?